docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,102 @@
+---
+CPS:
+Title: Sub-Optimal Provider to Wallet API
+Status: Open
+Category: Wallets
+Authors:
+    - Rhys Bartels-Waller <rhys.bartelswaller@iohk.io>
+    - William Wolff <william.wolff@iohk.io>
+    - Martynas Kazlauskas <martynas.kaslauskas@iohk.io>
+    - Daniele Ricci <daniele.ricci@iohk.io>
+    - Sefa Irken <sefa.irken@iohk.io>
+
+
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/tree/master/CPS-0011
+Created: 2024-07-09
+License: CC-BY-4.0
+---
+
+## Abstract
+The event-based nature of a Blockchain is well-suited for wallets to derive their internal state from genesis, however, most data services only expose a request/response interface and/or take on the responsibility of decoding the native binary data into JSON, enforcing a strong client/server contract with the current protocol transaction format. Wallets must use long polling to detect changes, on an interval that balances latency with network and service demand, then make numerous calls against various endpoints that have the potential to change on each protocol upgrade. Furthermore, the lack of an API standard inhibits development and adoption of a standalone desktop service for wallets to offer a full node experience for their users, which is best achieved via a universal API running either on a remote server, in a multi-tenant mode, or locally running in single-tenant mode.
+
+## Problem
+> [!NOTE]
+>
+> Fundamental wallet functionality is defined in this document as being able to construct and submit any transaction supported by the latest Cardano protocols, requiring:
+> - UTxO set owned by addresses derived within the 1852 purposed, [BIP32-ed25519](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf) accounts or the product of hashing a script.
+> - Active stake pools
+> - Reward account balances
+> - Certificates submitted by the wallet
+> - Current Cardano network protocol parameters
+> - Era summaries, for time conversions
```

Comment:
Not strictly required for fundamental functionality; If we add this, then might as well add entire tx history instead of just certificates submitted by the wallet
```suggestion
```

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/3f074c21d7f7748213637511c668634badb36562